### PR TITLE
feat(cli): check highlighting for consecutive carat `^` columns

### LINKED ISSUE #2854: Highlighting check doesn't verify multiple carat columns (`^`)
In the Sublime test specification, you should be able to repeat `^` characters, and each column gets checked. For example:

image

However, it doesn't seem like tree-sitter verifies this. If I add more carats so that the span should be invalid, the test still passes.

image

### PR DESCRIPTION
**Pull request by novusnota**
Hey there! I'm super new to Rust, but I've tried to preserve the code style in the remaining file and not to refactor any existing loops or logic. Just slightly adjusted it to make tree-sitter's highlighting tests actually be Sublime-like.

From [Tree-sitter docs on Unit testing of syntax highlighting](https://tree-sitter.github.io/tree-sitter/syntax-highlighting#unit-testing):

> Consecutive ^s will test each column against the selector.

Now that's the case :)

Closes #2854

P.S.: Failed checks are not on me — `tree-sitter-sway` seems to be failing, [filed an issue there](https://github.com/FuelLabs/tree-sitter-sway/issues/11)  (I hope I described things correctly :heart:)

### CODE REVIEW

**Comment by amaanq** [comment]
> P.S.: Failed checks are not on me — tree-sitter-sway seems to be failing, https://github.com/FuelLabs/tree-sitter-sway/issues/11 (I hope I described things correctly ❤️)

Yeah no worries on that, those specific failing checks aren't a hard requirement. This looks great, but my only question is what if the `Assertion` type stored a `size` field - this way in case we have lots of carats we aren't creating a new `Assertion` object for each individual carat. You would have to change some of the code where assertions are used though to account for this, though

**Comment by amaanq** [comment]
Yeah, I think it's worth it

**Review by 818Nawaf — APPROVED** [review]

**Review by 818Nawaf — APPROVED** [review]

**Comment by amaanq** [comment]
ping @novusnota

**Comment by novusnota** [comment]
Whoops, got distracted by other matters. Will be finishing the proposed refactor by this Wednesday.

**Comment by novusnota** [comment]
CI fails on something out of scope of my PR, idk what's the fix there.

**Comment by amaanq** [comment]
it's fine, you probably just need to rebase to fix it (but no worries)

File: cli/src/query_testing.rs [hunk 1]
```diff
@@ -117,15 +120,26 @@ pub fn parse_position_comments(
...
                         for (i, c) in text.char_indices() {
                             arrow_end = i + 1;
                             if c == '-' && has_left_caret {
                                 has_arrow = true;
+                                arrow_count = 1;
                                 break;
                             }
                             if c == '^' {
                                 has_arrow = true;
+                                arrow_count = 1;
```
[the remark this answers is not in the document]
Comment (novusnota) [reply cli/src/query_testing.rs]: Sure. My code explicitly set `arrow_count` to 1 only next to `has_arrow` just so one won't accidentally mess up the logic when returning to this later on. But let's go with your suggestion :)

**Review by amaanq — APPROVED** [review]

**Pull request merged**